### ISSUE TITLE
Question about Langevin dynamics sampling

### ISSUE THREAD

**Issue by TFNTF** [zhangbingliang2019/DAPS#3]
Can you please explain the following question:

In order to sample x_{0|y},  Langevin dynamics is applied. In Eq.(7) shown in the article, there are \nabla_{x_0^j}\log p(x_0^j|x_t) and \nabla_{x_0^j}\log p(y|x_0^j), but these two terms are disappeared in lgvd.sample of sampler.py in the submitted code.

Are these two terms are ignored in practice?

Thanks
![1](https://github.com/user-attachments/assets/54be35a5-e286-4446-b223-ab178194305c)
![2](https://github.com/user-attachments/assets/a82a3d56-a1f8-4396-bfc4-3a466e604c6f)

**Comment by chuwd19** [comment]
Thanks for asking. The `operator.error` in the code corresponds to the second term in Eq.(7), and the L2 loss in the code corresponds to the first term. The update is automatically handled by `optimizer.step()`. You can refer to Eq.(9) in our paper and compare it with the code. ($\beta_y$ in the equation is named as `self.tau`)
![image](https://github.com/user-attachments/assets/aedc218b-4eb4-488e-9a12-0d971cecd622)

**Comment by TFNTF** [comment]
Sincerely thanks to your in time response.
